src/managers/FileManager.ts

```diff
@@ -38,6 +37,70 @@ export class FileManager {
         return this.cache[lowerFilePath] as CodeFile;
     }
 
+    private getFunctionInfo(lines: string[]) {
+        let result = [];
+
+        let functionStack = [] as FunctionInfo[];
+        for (let lineIndex = 0; lineIndex < lines.length; lineIndex++) {
+            let line = lines[lineIndex];
+            let functionName: string;
+            let openers = [
```

Comment:
@chrisdp  @georgejecook Are there any other function declaration patterns that we should look for that I don't handle in this list?